File: app/src/main/java/jp/bellware/echo/main/MainCallback.java
package jp.bellware.echo.main;

/**
 * MainServiceからのコールバック
 */
public interface MainCallback {
    /**
     * 状態を更新する
     * @param animation アニメーションフラグ
     * @param status 状態
     */
    void onUpdateStatus(boolean animation, QRecStatus status);

    /**
     * 視覚的ボリュームを更新する
     * @param volume
     */
    void onUpdateVolume(float volume);

    /**
     * 警告メッセージ表示
     */
    void onShowWarningMessage(int resId);

}

File: app/src/main/java/jp/bellware/echo/main/MainCallback.kt
package jp.bellware.echo.main

/**
 * MainServiceからのコールバック
 */
interface MainCallback {
    /**
     * 状態を更新する
     * @param animation アニメーションフラグ
     * @param status 状態
     */
    fun onUpdateStatus(animation: Boolean, status: QRecStatus)

    /**
     * 視覚的ボリュームを更新する
     * @param volume
     */
    fun onUpdateVolume(volume: Float)

    /**
     * 警告メッセージ表示
     */
    fun onShowWarningMessage(resId: Int)

}
